src/local-storage-engine.spec.ts

```diff
@@ -0,0 +1,239 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import * as LZString from 'lz-string';
+
+import { LocalStorageEngine } from './local-storage-engine';
+
+describe('LocalStorageEngine Compression Migration', () => {
+  let mockLocalStorage: Storage;
+
+  beforeEach(() => {
+    mockLocalStorage = {
+      getItem: jest.fn(),
+      setItem: jest.fn(),
+      removeItem: jest.fn(),
+      clear: jest.fn(),
+      get length() {
+        return this._length || 0;
```

Comment:
minor but for forcing undefined numbers to `0` I find null coalesce (`??`) more precise.